src/org/authInfo.ts

```diff
@@ -440,6 +456,59 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
     return AuthInfo.listAllAuthorizations((possibleHub) => possibleHub?.isDevHub ?? false);
   }
 
+  private static async identifyPossibleSandbox(
+    possibleProdOrg: OrgAuthorization,
+    fields: AuthFields,
+    orgAuthInfo: AuthInfo,
+    logger: Logger
+  ): Promise<void> {
+    if (!fields.orgId) {
+      return;
+    }
+
+    try {
+      const prodOrg = await Org.create({ aliasOrUsername: possibleProdOrg.username });
+      const sbxProcess = await prodOrg.querySandboxProcessByOrgId(fields.orgId);
+      if (!sbxProcess?.SandboxInfoId) {
+        return;
+      }
+      logger.debug(`${fields.orgId} is a sandbox of ${possibleProdOrg.username}`);
+
+      try {
+        await orgAuthInfo.save({
+          ...fields,
+          isScratch: false,
+          isSandbox: true,
+        });
+      } catch (err) {
+        logger.debug(`error updating auth file for: ${orgAuthInfo.getUsername()}`, err);
+        throw err; // rethrow; don't want a sandbox config file with an invalid auth file
+      }
+
+      try {
+        // set the sandbox config value
+        const sfSandbox = {
+          sandboxUsername: fields.username,
+          sandboxOrgId: fields.orgId,
+          prodOrgUsername: possibleProdOrg.username,
+          sandboxName: sbxProcess.SandboxName,
+          sandboxProcessId: sbxProcess.Id,
+          sandboxInfoId: sbxProcess.SandboxInfoId,
+          timestamp: new Date().toISOString(),
+        } as SandboxFields;
```

Comment:
is the `SandboxFields` type wrong that timestamp isn't part of it?  Can that be fixed, or at least leave a comment about why the type assertion is necessary?